File: backend/api_server/pkg/configs/database_config.go
package configs

import (
	"os"
	"strconv"
	"time"
)

type DatabaseConfigResult struct {
	Type            string
	URL             string
	MaxOpenConns    int
	MaxIdleConns    int
	ConnMaxLifetime time.Duration
	ConnMaxIdleTime time.Duration
}

func DatabaseConfig() *DatabaseConfigResult {
	dbType := os.Getenv("DATABASE_TYPE")
	if dbType == "" {
		dbType = "mysql"
	}

	url := os.Getenv("DATABASE_URL")
	if url == "" {
		url = "root:password@tcp(localhost:3306)/dbname"
	}

	maxOpenConns := 10
	if val := os.Getenv("DATABASE_MAX_OPEN_CONNS"); val != "" {
		if v, err := strconv.Atoi(val); err == nil {
			maxOpenConns = v
		}
	}

	maxIdleConns := 5
	if val := os.Getenv("DATABASE_MAX_IDLE_CONNS"); val != "" {
		if v, err := strconv.Atoi(val); err == nil {
			maxIdleConns = v
		}
	}

	connMaxLifetime := time.Hour
	if val := os.Getenv("DATABASE_CONN_MAX_LIFETIME"); val != "" {
		if d, err := time.ParseDuration(val); err == nil {
			connMaxLifetime = d
		}
	}

	connMaxIdleTime := time.Minute
	if val := os.Getenv("DATABASE_CONN_MAX_IDLE_TIME"); val != "" {
		if d, err := time.ParseDuration(val); err == nil {
			connMaxIdleTime = d
		}
	}

	return &DatabaseConfigResult{
		Type:            dbType,
		URL:             url,
		MaxOpenConns:    maxOpenConns,
		MaxIdleConns:    maxIdleConns,
		ConnMaxLifetime: connMaxLifetime,
		ConnMaxIdleTime: connMaxIdleTime,
	}
}

File: backend/ai_server/ai_server/pkg/configs/database_config.py
import os
from dataclasses import dataclass

@dataclass
class DataBaseConfigResult:
    database_type: str
    database_host: str
    database_port: int
    database_user: str
    database_password: str
    database_name: str
    database_ssl_mode: str
    database_max_open_conns: int
    database_max_idle_conns: int
    database_conn_max_lifetime: str
    database_conn_max_idle_time: str

def get_database_config() -> DataBaseConfigResult:
    return DataBaseConfigResult(
        database_type=os.getenv("DATABASE_TYPE", "mysql"),
        database_host=os.getenv("DATABASE_HOST", "localhost"),
        database_port=int(os.getenv("DATABASE_PORT", 3306)),
        database_user=os.getenv("DATABASE_USER", "root"),
        database_password=os.getenv("DATABASE_PASSWORD", "password"),
        database_name=os.getenv("DATABASE_NAME", "api_server_db"),
        database_ssl_mode=os.getenv("DATABASE_SSL_MODE", "prefer"),
        database_max_open_conns=int(os.getenv("DATABASE_MAX_OPEN_CONNS", 100)),
        database_max_idle_conns=int(os.getenv("DATABASE_MAX_IDLE_CONNS", 10)),
        database_conn_max_lifetime=os.getenv("DATABASE_CONN_MAX_LIFETIME", "30"),
        database_conn_max_idle_time=os.getenv("DATABASE_CONN_MAX_IDLE_TIME", "30")
    )